src/OpenSearch/Common/Exceptions/BadMethodCallException.php

```diff
@@ -21,6 +21,7 @@
 
 namespace OpenSearch\Common\Exceptions;
 
+// @phpstan-ignore classConstant.deprecatedClass
 @trigger_error(BadMethodCallException::class . ' is deprecated in 2.3.2 and will be removed in 3.0.0.', E_USER_DEPRECATED);
```

Comment:
We could also remove these trigger_errors. Packages like https://packagist.org/packages/symfony/error-handler emit those based on `@deprecations` annotation 🤔

Reply:
I don't think we can rely on symfony/error-handler in our lib. I think that's something that an app would use?